main.ts

```diff
@@ -159,8 +159,10 @@ export default class HideFoldersPlugin extends Plugin {
     // used for re-processing folders when a folder is expanded in the file-navigator
     this.mutationObserver = new MutationObserver((mutationRecord) => {
       mutationRecord.forEach(record => {
-        if(record.target?.parentElement?.classList.contains("nav-folder")) {
-          this.processFolders();
+        if(record.target?.parentElement?.classList.contains("nav-files-container")) {
```

Comment:
See my comment in the PR-thread, add back the old check and just check if the mutationRecord's parent classList contains either `nav-folder` or `nav-files-container`